src/App.js

```diff
@@ -1,19 +1,141 @@
 import React from 'react';
 import './App.css';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+function getUserById(userId) {
+  return users.find(user => user.id === userId);
+}
+
+const todosWithUser = todos.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todosNew: todosWithUser,
+    todoTitle: '',
+    todoUserId: 0,
+    errorNoTitle: false,
+    errorNoUser: false,
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+        <p>
+          <span>Users: </span>
+          {users.length}
+        </p>
+        <form
+          action="/api/todolist"
+          method="Post"
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            this.setState((prevState) => {
+              if (prevState.todoTitle !== '' && prevState.todoUserId !== 0) {
+                const newTodo = {
+                  id: Math.random(),
```

Comment:
better to use the previous id + 1

Reply:
Changed. Have used new prevState.todosNew.length + 1 (when I use prevState.id + 1  -> become error in console about ununique key after second added todo)